src/test/regress/expected/with_modifying.out

```diff
@@ -25,52 +25,148 @@ SELECT
+       1 |                                 |         |         |      41 |        
+(9 rows)
+
 -- multi-shard UPDATE in CTE
 WITH basic_update AS (
 	UPDATE users_table SET value_3=42 WHERE value_2=1 RETURNING *
 )
 SELECT
 	*
 FROM
-	basic_update;
-ERROR:  data-modifying statements are not supported in the WITH clauses of distributed queries
+	basic_update
+ORDER BY
+	user_id,
+	time;
+ user_id |              time               | value_1 | value_2 | value_3 | value_4 
+---------+---------------------------------+---------+---------+---------+---------
+       2 | Thu Nov 23 06:50:30.797805 2017 |       1 |       1 |      42 |        
+       2 | Thu Nov 23 06:56:38.46819 2017  |       0 |       1 |      42 |        
+       2 | Thu Nov 23 13:52:54.83829 2017  |       3 |       1 |      42 |        
+       3 | Wed Nov 22 18:43:51.450263 2017 |       1 |       1 |      42 |        
+       3 | Thu Nov 23 00:15:45.610845 2017 |       1 |       1 |      42 |        
+       4 | Wed Nov 22 23:59:46.493416 2017 |       3 |       1 |      42 |        
+       4 | Thu Nov 23 01:55:21.824618 2017 |       3 |       1 |      42 |        
+       4 | Thu Nov 23 06:50:08.101207 2017 |       2 |       1 |      42 |        
+       4 | Thu Nov 23 07:09:37.382372 2017 |       4 |       1 |      42 |        
+       4 | Thu Nov 23 08:38:45.877401 2017 |       4 |       1 |      42 |        
+       4 | Thu Nov 23 11:22:36.283022 2017 |       4 |       1 |      42 |        
+       4 | Thu Nov 23 12:59:23.28005 2017  |       4 |       1 |      42 |        
+       4 | Thu Nov 23 13:56:39.967304 2017 |       0 |       1 |      42 |        
+       4 | Thu Nov 23 14:57:36.278713 2017 |       3 |       1 |      42 |        
+       5 | Wed Nov 22 22:31:47.62577 2017  |       3 |       1 |      42 |        
+       5 | Thu Nov 23 06:36:11.898621 2017 |       5 |       1 |      42 |        
+       5 | Thu Nov 23 09:17:14.659809 2017 |       3 |       1 |      42 |        
+       5 | Thu Nov 23 10:46:56.036776 2017 |       3 |       1 |      42 |        
+       5 | Thu Nov 23 14:29:02.557934 2017 |       2 |       1 |      42 |        
+       6 | Wed Nov 22 20:15:53.317797 2017 |       1 |       1 |      42 |        
+       6 | Thu Nov 23 00:07:11.068353 2017 |       1 |       1 |      42 |        
+(21 rows)
+
 -- single-shard DELETE in CTE
 WITH basic_delete AS (
-	DELETE FROM users_table WHERE user_id=42 RETURNING *
+	DELETE FROM users_table WHERE user_id=6 RETURNING *
 )
 SELECT
 	*
 FROM
 	basic_delete;
-ERROR:  data-modifying statements are not supported in the WITH clauses of distributed queries
+ user_id |              time               | value_1 | value_2 | value_3 | value_4 
+---------+---------------------------------+---------+---------+---------+---------
+       6 | Thu Nov 23 00:09:44.19812 2017  |       5 |       2 |       0 |        
+       6 | Thu Nov 23 14:43:18.024104 2017 |       3 |       2 |       5 |        
```

Comment:
These could use some ORDER BY's

Reply:
Good catch, I added some, and I will also check others. I will also do a bit more testing.